api/ingest-tokens.go

```diff
@@ -79,29 +79,46 @@ func toIngestToken(data ingestTokenData) *IngestToken {
 }
 
 func (i *IngestTokens) Add(repositoryName string, tokenName string, parserName string) (*IngestToken, error) {
-	var mutation struct {
-		Result struct {
-			IngestToken ingestTokenData
-		} `graphql:"addIngestToken(repositoryName: $repositoryName, name: $tokenName, parser: $parserName)"`
+	var parserIdArg *graphql.String
+	if parserName != "" {
+		parser, err := i.client.Parsers().Get(repositoryName, parserName)
+		if err != nil {
+			return nil, fmt.Errorf("unable to look up parser id for parser name %q: %w", parserName, err)
+		}
+		parserIdArg = graphql.NewString(graphql.String(parser.Id))
 	}
 
-	var parserNameArg *graphql.String
-	if parserName != "" {
-		parserNameArg = graphql.NewString(graphql.String(parserName))
+	var mutation struct {
+		IngestToken struct {
+			Name   string
+			Token  string
+			Parser *struct {
+				Name string
+			}
+		} `graphql:"addIngestTokenV2(input: { repositoryName: $repositoryName, name: $tokenName, parserId: $parserId})"`
 	}
 
 	variables := map[string]interface{}{
 		"tokenName":      graphql.String(tokenName),
 		"repositoryName": graphql.String(repositoryName),
-		"parserName":     parserNameArg,
+		"parserId":       parserIdArg,
 	}
 
 	err := i.client.Mutate(&mutation, variables)
 	if err != nil {
 		return nil, err
 	}
 
-	return toIngestToken(mutation.Result.IngestToken), err
+	ingestToken := IngestToken{
+		Name:  mutation.IngestToken.Name,
+		Token: mutation.IngestToken.Token,
+	}
+
+	if mutation.IngestToken.Parser != nil {
+		ingestToken.AssignedParser = mutation.IngestToken.Parser.Name
+	}
+
+	return &ingestToken, err
```

Comment:
```suggestion
	return &ingestToken, nil
```

err is always nil here.